src/modules/commander/Commander.cpp

```diff
@@ -4058,6 +4068,164 @@ void Commander::battery_status_check()
 	}
 }
 
+void Commander::airspeed_use_check()
+{
+	// assume airspeed sensor is good before starting FW flight
+	bool valid_flight_condition = (status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) &&
+			!status.is_rotary_wing &&
+			!land_detector.landed;
+	bool fault_declared = false;
+	bool fault_cleared = false;
+	if (!valid_flight_condition) {
+		_tas_use_inhibit = false;
+		_time_tas_good_declared = hrt_absolute_time();
+		_time_tas_bad_declared = 0;
+		status.aspd_check_failing = false;
+		status.aspd_fault_declared = false;
+		status.aspd_use_inhibit = false;
+		status.aspd_fail_rtl = false;
+		_time_last_airspeed = hrt_absolute_time();
+		_airspeed_fault_type = new(char[7]);
+	} else {
+		// The vehicle is flying so use the status of the airspeed innovation check '_tas_check_fail' in
+		// addition to a sanity check using airspeed and load factor and a missing sensor data check.
+
+		// Check if sensor data is missing - assume a minimum 5Hz data rate.
+		bool data_missing = false;
+		if ((hrt_absolute_time() -_time_last_airspeed) > 200000) {
+			data_missing = true;
+		}
+
+		// Declare data stopped if not received for longer than 1 second
+		bool data_stopped = false;
+		if ((hrt_absolute_time() -_time_last_airspeed) > 1000000) {
+			data_stopped = true;
+		}
+		_time_last_airspeed = hrt_absolute_time();
+
+		// Check if the airpeed reading is lower than physically possible given the load factor
+		_airspeed_sub.update();
+		_airspeed = _airspeed_sub.get();
+		_sensor_combined_sub.update();
+		_sensor_combined = _sensor_combined_sub.get();
+		float max_lift_ratio = _airspeed.indicated_airspeed_m_s /fmaxf(_airspeed_stall.get(),1.0f);
+		max_lift_ratio *= max_lift_ratio;
+		status.load_factor_ratio = 0.95f * status.load_factor_ratio - 0.05f * (_sensor_combined.accelerometer_m_s2[2] / 9.80665f) / max_lift_ratio;
```

Comment:
Just curious, but why not use az from vehicle_local_position?

Or if you want to use sensor data directly (rather than the derivative of velocity) what about the estimated bias?

Reply:
It needs to be body frame, not earth frame to capture wing generated specific force.

Reply:
In that case, using the bias corrected bodyframe acceleration from sensor_bias would be more appropriate, I think.

Reply:
Will fix. Also need to use absolute value of Z accel to allow for future inverted flight capability.